addon/components/hyper-table-v2/index.hbs

```diff
@@ -62,17 +67,15 @@
               {{#if this.features.selection}}
                 <div class="hypertable__column hypertable__column--selection">
                   <header>
-                    <OSS::Checkbox @checked={{eq @handler.selection "all"}} @size="sm"
-                                   @onChange={{this.toggleSelectAll}}
-                                   @disabled={{false}} />
+                    <OSS::Checkbox @checked={{gt this.selectionCount 0}} @size="sm" @onChange={{this.toggleSelectAll}}
+                                   @partial={{lt this.selectionCount @handler.rowsMeta.total}} />
                   </header>
 
                   {{#each @handler.rows as |row|}}
                     <HyperTableV2::Cell @handler={{@handler}} @column={{column}} @row={{row}}
                                         @onClick={{this.onRowClick}} @onHover={{this.onRowHover}}>
-                      <OSS::Checkbox @checked={{or (array-includes @handler.selection row) (eq @handler.selection "all")}}
-                                     @size="sm" @onChange={{fn this.toggleRowSelection row}}
-                                     @disabled={{eq @handler.selection "all"}} />
+                      <OSS::Checkbox @checked={{if (eq @handler.selection "all") (not (array-includes @handler.exclusion row)) (array-includes @handler.selection row)}}
```

Comment:
isn't this condition missing an `or` or an `and` ? 🤔 

Hard to read as is, I think this should be defined in the controller instead

Reply:
No `and` or `or` missing, I need to create a helper to do that

Reply:
My bad, i forgot about the "else" part on ifs in ember templates